src/renderer/components/app.tsx

```diff
@@ -86,6 +88,15 @@ export class App extends React.Component {
 
     await requestMain(clusterSetFrameIdHandler, clusterId);
     await getHostedCluster().whenReady; // cluster.activate() is done at this point
+
+    const activeEntityDisposer = reaction(() => catalogEntityRegistry.getById(clusterId), (entity) => {
+      if (!entity) {
+        return;
+      }
+      catalogEntityRegistry.activeEntity = entity;
+      activeEntityDisposer();
+    }, {fireImmediately: true});
```

Comment:
Why not just override `activeEntity` with a getter?

```suggestion
    Object.defineProperty(catalogEntityRegistry, "activeEntity", {
      get() {
        return catalogEntityRegistry.getById(clusterId);
      },
    });
```

Reply:
Actually for that matter, why is it not a getter for that class to begin with?